configuration/resize-disk-image.md

```diff
@@ -1,47 +1,72 @@
 ---
 layout: doc
-title: Resize Disk Image
+title: Resize Private Disk Image
 permalink: /doc/resize-disk-image/
 redirect_from:
 - /en/doc/resize-disk-image/
 - /doc/ResizeDiskImage/
 - /wiki/ResizeDiskImage/
 ---
 
-Resize Disk Image
+Resize Private Disk Image
 -----------------
 
-There are several disk images which can be easily extended. But pay attention to the overall consumed space of your sparse disk images.
+There are several disk images which can be easily extended, but pay attention to the overall consumed space of your sparse disk images.
+See also additional information and caveats about [resizing the root disk image](/doc/resize-root-disk-image/).
 
-### Private disk image
 
-1048576 MB is the maximum size which can be assigned to a private storage through qubes-manager.
+### Private disk image (R4.0)
 
-To grow the private disk image of a AppVM beyond this limit [qubes-grow-private](/doc/dom0-tools/qvm-grow-private/) can be used:
+1048576 MiB is the maximum size which can be assigned to private storage through Qube Manager.
+
+To grow the private disk image of an AppVM beyond this limit, `qvm-volume` can be used:
 
 ~~~
-qvm-grow-private <vm-name> <size>
+qvm-volume extend <vm_name>:private <size>
 ~~~
 
 Note: Size is the target size (i.e. 4096MB or 16GB, ...), not the size to add to the existing disk.
 
-### Shrinking private disk image (Linux VM)
+### Private disk image (R3.2)
+
+1048576 MB is the maximum size which can be assigned to private storage through Qubes Manager.
+
+To grow the private disk image of an AppVM beyond this limit, [qvm-grow-private](/doc/dom0-tools/qvm-grow-private/) can be used:
+
+~~~
+qvm-grow-private <vm-name> <size>
+~~~
+
+Note: Size is the target size (i.e. 4096MB or 16GB, ...), not the size to add to the existing disk. 
+
+### Shrinking private disk image (Linux VM, R4.0)
+
+1.  Create a new qube with smaller disk using Qube Manager or `qvm-create`
+2.  Move data to the new qube using `qvm-copy` or OS utilities
+3.  Delete old qube using Qube Manager or `qvm-remove`
```

Comment:
This procedure will also work on R3.2, is less risky and simpler. I suggest deprecating the R3.2 instruction and having this one for both versions.

Reply:
It is also possible to prepare similar (the risky one) instruction for R4.0, which do not require duplicating the data. But as said before, I'm not sure if that's a good idea.

Reply:
Will address on the next round!